qa/rpc-tests/test.py

```diff
@@ -0,0 +1,235 @@
+elif HOST_OS == 'Darwin':
+    os.environ["BITCOINCLI"] = WORK_DIR + r"\zcash-cli"
+    os.environ["BITCOIND"] = WORK_DIR + r"\zcashd"
+else:
+    logging.error(" %s is not currently supported.", HOST_OS)
+
+
+BASE_SCRIPTS=(
+    'paymentdisclosure.py',
+    'prioritisetransaction.py',
+    'wallet_treestate.py',
+    'wallet_anchorfork.py',
+    'wallet_changeindicator.py',
+    'wallet_import_export.py',
+    'wallet_protectcoinbase.py',
+    'wallet_shieldcoinbase_sprout.py',
+    'wallet_shieldcoinbase_sapling.py',
+    'wallet_listreceived.py',
+    'wallet.py',
+    'wallet_overwintertx.py',
+    'wallet_persistence.py', 
+    'wallet_nullifiers.py',
+    'wallet_1941.py',
+    'wallet_addresses.py',
+    'wallet_sapling.py',
+    'wallet_listnotes.py',
+    'mergetoaddress_sprout.py',
+    'mergetoaddress_sapling.py',
+    'listtransactions.py',
+    'mempool_resurrect_test.py',
+    'txn_doublespend.py',
+    'txn_doublespend.py --mineblock',
+    'getchaintips.py',
+    'rawtransactions.py',
+    'rest.py',
+    'mempool_spendcoinbase.py',
+    'mempool_reorg.py',
+    'mempool_tx_input_limit.py',
+    'mempool_nu_activation.py',
+    'mempool_tx_expiry.py',
+    'httpbasics.py',
+    'zapwallettxes.py',
+    'proxy_test.py',
+    'merkle_blocks.py',
+    'fundrawtransaction.py',
+    'signrawtransactions.py',
+    'signrawtransaction_offline.py',
+    'walletbackup.py',
+    'key_import_export.py',
+    'nodehandling.py',
+    'reindex.py',
+    'decodescript.py',
+    'blockchain.py',
+    'disablewallet.py',
+    'zcjoinsplit.py',
+    'zcjoinsplitdoublespend.py',
+    'zkey_import_export.py',
+    'reorg_limit.py',
+    'getblocktemplate.py',
+    'bip65-cltv-p2p.py',
+    'bipdersig-p2p.py',
+    'p2p_nu_peer_management.py',
+    'rewind_index.py',
+    'p2p_txexpiry_dos.py',
+    'p2p_txexpiringsoon.py',
+    'p2p_node_bloom.py',
+    'regtest_signrawtransaction.py',
+    'finalsaplingroot.py'
+)
+
+EXTENDED_SCRIPTS=(
+    'getblocktemplate_longpoll.py',
+    'getblocktemplate_proposals.py',
+    'pruning.py',
+    'forknotify.py',
+    'hardforkdetection.py',
+    'invalidateblock.py',
+    'keypool.py',
+    'receivedby.py',
+    'rpcbind_test.py',
+#   'script_test.py',
+    'smartfees.py',
+    'maxblocksinflight.py',
+    'invalidblockrequest.py',
+#    'forknotify.py',
+    'p2p-acceptblock.py'
+)
+
+NON_SCRIPTS =(
+    'rpc-tester.py'
+)
+
+def cleanup_failed_tests():
+    pass
+
+def run_test_scripts(test_list, args):
+    fail_list = []
+    pass_list = []
+    stats = []
+
+    total_tests = len(test_list)
+    num_tests_ran = 0
+
+    logging.info("=== Starting RPC Tester  ===")
+    total_stime = time.time()
+    for file in test_list:
+        logging.info("--- Running test : %s ---", file)
+        stats.append(file)
+        #might want to add routine to properly clean SIGTERM tests so they don't conga line the rest
+        try:
+            start_time = time.time()
+            p1 = Popen(['python', file])
```

Comment:
This leaves off various arguments and flags that are supposed to be provided to the individual RPC tests. See [our current test runner](https://github.com/zcash/zcash/blob/master/qa/pull-tester/rpc-tests.sh#L131-L134), as well as [upstream's Python test runner as of 0.12.0](https://github.com/bitcoin/bitcoin/blob/188ca9c305d3dd0fb462b9d6a44048b1d99a05f3/qa/pull-tester/rpc-tests.py#L144-L161).

Additionally, executing with `python` ignores the environment command at the top of each file. This might be useful for a CI system, but probably isn't what is indended for general usage, so if the intent is the latter then this should probably be turned on with a CLI flag.

Reply:
Still working through some of the upstream nuances, but it appears the defaults are redundant to the env `BITCOIND` and `BITCOINCLI`. Perhaps there are other binaries upstream uses inside of what they user for `srcdir`, but from what I have tinkered with it doesn't seem we use other binaries in the src dir.

Reply:
It also appears that the other major upgrade is the coverage python file. I'd have to go look at what it does in the field. At a glance it seems  to just dump pathways of control flow from RPC, with a given set of params.